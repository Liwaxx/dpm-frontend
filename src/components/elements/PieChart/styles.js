
const styles = {
  legends: {
    paddingLeft: 10,
    height:90
  },
  card: {
    maxWidth: 220,
    maxHeigh: 200
  },
  bullet: {
    display: 'inline-block',
    margin: '0 2px',
    transform: 'scale(0.8)',
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
  slices: [
    {
      color: '#A6EDE9'

    },
    {
      color: '#A8CEEB'
    },
    {
      color: '#FA8E8E'
    }
  ],
  chartArea: {
    left: 25,
    top: 10,
    right: 25,
    bottom: 10,
    width: 200,
    height: 155
  },
  fontName: 'Roboto',
  legend: 'none',
  pieSliceText: 'value',
  width: '170',
  height: '140',
  pieStartAngle: 90,
  tooltip: { trigger: 'none' },
  span2: {
    height: '12px', 
    width: '12px', 
    backgroundColor: '#A6EDE9', 
    borderRadius: '50%', 
    display: 'inline-block',
    marginRight:8,
    marginTop:2
  },
  span1: {
    height: '12px', 
    width: '12px', 
    backgroundColor: '#A8CEEB', 
    borderRadius: '50%', 
    display: 'inline-block',
    marginRight:8,
    marginTop:2
  },
  span0: {
    height: '12px', 
    width: '12px', 
    backgroundColor: '#FA8E8E', 
    borderRadius: '50%', 
    display: 'inline-block',
    marginRight:8,
    marginTop:2
  },
};


export default styles;