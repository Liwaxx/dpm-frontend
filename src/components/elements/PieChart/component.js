import React from 'react';
import PropTypes from 'prop-types';
import { Chart } from 'react-google-charts';
import Text from '../Text';
import styles from './styles';

export default class Component extends React.Component {
  render() {
    const { classes, legend } = this.props;

    return (
      <div>
        <Chart
          width={'100'}
          height={'100'}
          chartType="PieChart"
          loader={<div>Loading Chart</div>}
          data={this.props.dataChart}
          options={
            styles
          }
        />
        <div className={classes.legends}>
          <Text type="paragraph" style={{ paddingBottom: 10, paddingTop: 5 }}>Total Backlog</Text>
          {
            legend ? (
              legend.map((item, index) => (
                <Text type="subheader" key={index}>
                  <span className={classes[`span${index}`]} />
                  {item}
                </Text>
              ))
            ) : ''}
        </div>
      </div>
    );
  }
}

Component.propTypes = {
  classes: PropTypes.object,
  legend: PropTypes.array,
  dataChart: PropTypes.array
};