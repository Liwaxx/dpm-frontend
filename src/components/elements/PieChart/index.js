import { withStyles } from '@material-ui/core/styles';
import PieChart from './component';
import styles from './styles';

export default withStyles(styles)(PieChart);