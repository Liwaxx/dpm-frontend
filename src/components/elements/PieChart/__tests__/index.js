import React from 'react';
import renderer from 'react-test-renderer';
import PieChart from '../index';

jest.mock('../../Text');
jest.mock('../styles');

describe('PieChart', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<PieChart />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with legend', () => {
    const legend = ['tes'];
    const tree = renderer
      .create(<PieChart legend={legend} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});