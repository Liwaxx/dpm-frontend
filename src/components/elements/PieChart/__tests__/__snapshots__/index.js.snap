// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PieChart renders correctly 1`] = `
<div>
  <div>
    Loading Chart
  </div>
  <div
    className="Component-legends-1"
  >
    
  </div>
</div>
`;

exports[`PieChart renders with legend 1`] = `
<div>
  <div>
    Loading Chart
  </div>
  <div
    className="Component-legends-1"
  />
</div>
`;
